CMakeLists.txt

```diff
@@ -183,7 +183,10 @@ include( TargetArch )
 target_architecture(ZDOOM_TARGET_ARCH)
 
 if( ${ZDOOM_TARGET_ARCH} MATCHES "x86_64" )
-	set( HAVE_VM_JIT ON )
+	# To make the game respecting OpenBSD's W^X policy by default
+	if(NOT CMAKE_PLATFORM_NAME MATCHES "OpenBSD")
+		set( HAVE_VM_JIT ON )
+	endif()
```

Comment:
In my opinion, AsmJit needs to be adapted to such memory protection policy instead of disabling JIT completely.
Most likely, [this fragment](https://github.com/coelckers/gzdoom/blob/e0fb9a45e21daba6afa40ef192aa7d9e3923ef20/libraries/asmjit/asmjit/base/osutils.cpp#L134-L138) should be extended to never have write and execute permissions at the same time.

Reply:
I read somewhere in here it is already a heavily modified version of asmjit, might be better some upstreaming job done first ?

Reply:
Upstreaming our asmjit changes will most likely not happen. The upstream author doesn't seem particular interested in the features in question (or he's too busy with other things to deal with them) and I don't think I can create a proper PR that would be accepted upstream.